snap7/client/__init__.py

```diff
@@ -97,7 +101,7 @@ def destroy(self) -> Optional[int]:
         logger.info("destroying snap7 client")
         if self._lib and self._s7_client is not None:
             return self._lib.Cli_Destroy(byref(self._s7_client))
-        self._s7_client = None
+        self._s7_client = None  # type: ignore[assignment]
```

Comment:
so `self._s7_client` is actually optional (can be None), you removed that above and now you need to type ignore that.

Reply:
If the type of `self._s7_client` is annotated as `Optional`, then `mypy` requires a check that `self._s7_client is not None` before each usage.`self._s7_client` is only None after the object is destroyed. So I decided to make the type S7Object and suppress the mypy error in just one place, instead of adding checks in many places.

![image](https://github.com/gijzelaerr/python-snap7/assets/52915342/e04a17e3-f979-4589-af6a-be55a2884cd0)


Do we really need the `self._s7_client` to be able to be None?

Reply:
this is just a choice, do you want to follow mypy strictly or do you trust your overrides? I don't mind some None checking.

Reply:
ok lets just do this, its cleaner and it doesn't matter that much